### PR TITLE
Break copying files into its own method

### PR DESCRIPTION
**Pull request by jazzsequence**
This QoL PR adds two functions (one of which is public and can be run as a `demigod` command itself).

* `getFramework` (`demigod:get-framework`) pulls the site framework out of `site:info` and returns the result. This is one of the things that would be required for adding a WordPress integration.
* `copyFrameworkFiles` puts all the logic handling the actual copying of files into its own function, depending on the framework. This simplifies the code, reinforces single-responsibility principle in `copyTemplates`, and allows that logic to extend to include options for WordPress sites

### CODE REVIEW

File: src/Commands/CopyTemplatesCommand.php [hunk 1]
```diff
@@ -53,46 +53,82 @@ public function copyTemplates(string $site_name) {
...
+  /**
+   * Get the site framework.
+   *
+   * @command demigod:get-framework
+   *
+   * @return string The site framework, pulled from site:info.
+   */
+  public function getFramework() : string {
+    $output = [];
+    exec( 'terminus site:info | grep Framework | xargs', $output );
```
Comment (stovak) [inline src/Commands/CopyTemplatesCommand.php]: This command will actually only give you the site info for the first site in your list of sites. try `terminus site:info ${PROJECT_NAME} --format=json | jq -r .framework `

File: src/Commands/CopyTemplatesCommand.php [hunk 2]
```diff
@@ -53,46 +53,82 @@ public function copyTemplates(string $site_name) {
...
+   * Copy the template files based on the framework.
+   *
+   * @param string $framework The CMS framework identified by getFramework().
+   * @param array ...$args Array of arguments required for copying files.
+   *              $site_name The site name that was called into copyTemplates.
+   *              $base_dir This plugin's base directory.
+   *              $clone_dir The directory the site was cloned into.
+   */
+  private function copyFrameworkFiles( string $framework, ...$args ) {
+    list( $site_name, $base_dir, $clone_dir ) = $args;
```
Comment (stovak) [inline src/Commands/CopyTemplatesCommand.php]: ❤️ this.

Comment (stovak) [reply src/Commands/CopyTemplatesCommand.php]: Also, not sure if you know this, but as of 8.0 you can deconstruct the array the same way you would in Javascript `[$one, $two, $three] = $array;`

File: templates/wordpress/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    expose:
+      - 80
+      - 9222
+    depends_on:
+      - php
+      - mysql
+    networks:
+      - demigod
+    volumes:
+      - ".:/var/www"
```
Comment (stovak) [inline templates/wordpress/docker-compose.yml:16]: Two things here: 1. Does the nginx virtual host config need to be different for WP than it does for durple? If so, we could inject it.  2. We need some way of sniffing to make sure WP and durple have `webroot true` in their pantheon.yml.

Comment (stovak) [reply templates/wordpress/docker-compose.yml:16]: FYI: https://github.com/demigod-tools/demigods-nginx/tree/master/nginx

**Review by stovak — CHANGES_REQUESTED** [review]

File: src/Commands/CopyTemplatesCommand.php (hunk 2 above)
Comment (stovak) [reply src/Commands/CopyTemplatesCommand.php]: This is an FYI. All this is fine. No changes.

File: templates/wordpress/docker-compose.yml (hunk 3 above)
Comment (jazzsequence) [reply templates/wordpress/docker-compose.yml:16]: this docker-compose is really just a draft and mostly copy-pasta of the drupal one. a WordPress specific docker-compose (and other template files) will come in #3

File: src/Commands/CopyTemplatesCommand.php (hunk 2 above)
Comment (jazzsequence) [reply src/Commands/CopyTemplatesCommand.php]: 🤯  
I think I either didn't realize you could deconstruct spreads or I had forgotten but this is amazing. ‼️

**Review by stovak — APPROVED** [review]

**Comment by stovak** [comment]
Accidentally merged this in and had to revert it.

**Comment by jazzsequence** [comment]
What broke?

**Pull request merged**